### ISSUE TITLE
Enhancement: Make ZenjectBinding extendable

### ISSUE THREAD

**Issue by JLeybovich** [ssannandeji/Zenject-2019#364]
We have a need to extend ZenjectBinding to provide an Identifier that is not a string (regular object). Would it be possible to change ZenjectBinding.Identifier from string to object and have the getters marked as virtual? The end goal is to use Types instead of strings to be more secure which seems doable based on that Id can be any object.

Thanks,

**Comment by svermeulen** [comment]
Yeah that change makes sense to me.  I'd be happy to merge a pull request like that.
